cognite/client/data_classes/hosted_extractors/mappings.py

```diff
@@ -0,0 +1,292 @@
+        self.input = input
+        self.created_time = created_time
+        self.last_updated_time = last_updated_time
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            external_id=resource["externalId"],
+            mapping=CustomMapping._load(resource["mapping"]),
+            published=resource["published"],
+            input=InputMapping._load(resource["input"]),
+            created_time=resource["createdTime"],
+            last_updated_time=resource["lastUpdatedTime"],
+        )
+
+    def as_write(self) -> MappingWrite:
+        return MappingWrite(
+            external_id=self.external_id, mapping=self.mapping, published=self.published, input=self.input
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case)
+        output["input"] = self.input.dump(camel_case)
+        return output
+
+
+class MappingWriteList(CogniteResourceList[MappingWrite], ExternalIDTransformerMixin):
+    _RESOURCE = MappingWrite
+
+
+class MappingList(WriteableCogniteResourceList[MappingWrite, Mapping], ExternalIDTransformerMixin):
+    _RESOURCE = Mapping
+
+    def as_write(self) -> MappingWriteList:
+        return MappingWriteList([mapping.as_write() for mapping in self.data])
+
+
+class MappingUpdate(CogniteUpdate):
+    def __init__(self, external_id: str) -> None:
+        super().__init__(external_id=external_id)
+
+    class _MappingUpdate(CognitePrimitiveUpdate):
+        def set(self, value: CustomMapping) -> MappingUpdate:
+            return self._set(value.dump())
+
+    class _InputMappingUpdate(CognitePrimitiveUpdate):
+        def set(self, value: InputMapping | None) -> MappingUpdate:
+            return self._set(value.dump() if value is not None else None)
+
+    class _PublishedUpdate(CognitePrimitiveUpdate):
+        def set(self, value: bool) -> MappingUpdate:
+            return self._set(value)
+
+    @property
+    def mapping(self) -> MappingUpdate._MappingUpdate:
+        return self._MappingUpdate(self, "mapping")
+
+    @property
+    def input(self) -> MappingUpdate._InputMappingUpdate:
+        return self._InputMappingUpdate(self, "input")
+
+    @property
+    def published(self) -> MappingUpdate._PublishedUpdate:
+        return self._PublishedUpdate(self, "published")
+
+    @classmethod
+    def _get_update_properties(cls, item: CogniteResource | None = None) -> list[PropertySpec]:
+        return [
+            PropertySpec("mapping", is_nullable=False),
+            PropertySpec("input", is_nullable=True),
+            PropertySpec("published", is_nullable=False),
+        ]
+
+
+_INPUTMAPPING_CLASS_BY_TYPE: dict[str, type[InputMapping]] = {
+    subclass._type: subclass  # type: ignore[type-abstract]
+    for subclass in InputMapping.__subclasses__()
+    if hasattr(subclass, "_type")
```

Comment:
Better to let this fail if a developer adds an invalid `InputMapping`:
```suggestion
    for subclass in InputMapping.__subclasses__()
```